File: Program42.cpp
#include<iostream>
using namespace std;

class Factor
{
    private:
        int iNo;
    
    public:
        Factor(int iValue)
        {
            this->iNo = iValue;
        }
        bool CheckPrime()
        {
            int iCnt = 0;
            
            for(iCnt = 2; iCnt <= (iNo/2); iCnt++)
            {
                if((iNo % iCnt) == 0)
                {
                    break;
                }
            }

            if(iCnt == (iNo/2)+1)
            {
                return true;
            }
            else
            {
                return false;
            }
        }
};

int main()
{
    int iValue = 0;
    bool bRet;

    cout<<"Enter number"<<endl;
    cin>>iValue;

    Factor fobj(iValue);
    
    bRet = fobj.CheckPrime();

    if(bRet == true)
    {
        cout<<iValue;
        cout<<" is a prime number";
    }
    else
    {
        cout<<iValue;
        cout<<" is not a prime number";
    }
    return 0;
}

File: Program42.java
import java.lang.*;
import java.util.*;

class Factors
{
     private int iNo;

     public void Accept()
     {
         Scanner sobj = new Scanner(System.in);
         System.out.println("Enter number");
         this.iNo = sobj.nextInt();
     } 
     public void Display()
     {
         System.out.println("Entered Value is: "+this.iNo);
     }
     public Boolean CheckPrime()
     {
         int iCnt = 0;
         
         for(iCnt = 2; iCnt <= (iNo/2); iCnt++)
         {
             if((iNo % iCnt) == 0)
             {
                 break;
             }
         }
         if(iCnt == (iNo/2)+1)
         {
              return true;
         }
         else
         {
             return false;
         }
     }
}

class Program42
{
    public static void main(String arg[])
    {
       boolean bRet;
       Factors fobj = new Factors();
       
       fobj.Accept();
       fobj.Display();
       
       bRet = fobj.CheckPrime();

       if(bRet == true)
       {
           System.out.println("Prime number");
       }
       else
       {
           System.out.println("Not prime number");
       }
    }
}